CV Backend/src/main/java/com/qa/demo/Interoperability/AccountController.java

```diff
@@ -0,0 +1,233 @@
+        System.out.println("Account creation attempting");
+        account.set_id(ObjectId.get());
+        //account.setUserRole("user");
+        //System.out.println(bCode.encode(account.getPassword()));
+        account.setPassword(bCode.encode(account.getPassword()));
+        repository.save(account);
+        return account;
+    }
+
+//    @DeleteMapping(value = "/delete/{id}")
+//    public String deleteFileFromUser(@PathVariable("id") ObjectId id){
+//
+//    }
+
+    @PostMapping(value = "/upload/{id}")
+    public String addFileToUser(@RequestParam("file") MultipartFile multipartFile, @PathVariable("id") ObjectId id){
+        System.out.println("File uploading being attempted");
+        Account account = repository.findBy_id(id);
+        try {
+            mongoTemplate.upsert(Query.query(where("_id").is(id)), Update.update("file",new Binary(BsonBinarySubType.BINARY, multipartFile.getBytes())), Account.class);
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "File addition failed";
+        }
+        return "File addition successful to account:"+account.getFirstName()+" "+account.getLastName();
+    }
+
+    @GetMapping(value = "/download/{id}")
+    public ResponseEntity<byte[]> getUserFile(@PathVariable("id") ObjectId id){
+        System.out.println("File download being attempted");
+        Account account = repository.findBy_id(id);
+        Binary accountCV = repository.findBy_id(id).getFile();
+        FileOutputStream fileOutputStream = null;
+        if(accountCV != null){
+
+            try {
+                fileOutputStream = new FileOutputStream("CV.pdf");
+                fileOutputStream.write(accountCV.getData());
+
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
+                System.out.println("File does not exist for user");
+            } catch (IOException e) {
+                e.printStackTrace();
+            } finally {
+                if(fileOutputStream != null){{
+                    try {
+                        fileOutputStream.close();
+                    }
+                    catch (IOException e) {
+                        e.printStackTrace();
+                        System.out.println("File stream could not be closed");
+                    }
+                }
+                }
+            }
+        }
+        HttpHeaders headers = new HttpHeaders();
+        //headers.setContentType(MediaType.valueOf(accountCV.getContentType()));
+        headers.setContentType(MediaType.APPLICATION_PDF);
+        headers.setContentLength(accountCV.getData().length);
+        headers.set("Content-Disposition", "attachment; filename="+account.getFirstName()+account.getLastName()+".pdf");
+        return new ResponseEntity<>(accountCV.getData(), headers, HttpStatus.OK);
+    }
+
+
+    @GetMapping(value = "/retrieve/{id}")
+    public String retrieveFileFromUser(@PathVariable("id") ObjectId id){
```

Comment:
same issues with this method like with the previous ones, I won't repeat myself just address them based on previous comments

Reply:
done